src/checkers/inference/solver/util/PrintUtils.java

```diff
@@ -250,6 +251,15 @@ public Void serialize(PreferenceConstraint preferenceConstraint) {
             return null;
         }
 
+        @Override
+        public Void serialize(ImplicationConstraint implicationConstraint) {
+            for (Constraint assumption : implicationConstraint.getAssumptions()) {
+                assumption.serialize(this);
```

Comment:
should there be " & " printed between each assumption? can you add a sample of the output from debug solver to the PR overview?

Reply:
Done